modules/jupyternotebook/main.nf

```diff
@@ -0,0 +1,87 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+include { dump_params_yml } from "./parametrize"
+
+params.options = [:]
+options        = initOptions(params.options)
+params.parametrize = true
+params.implicit_params = true
+params.meta_params = true
+
+process JUPYTERNOTEBOOK {
+    tag "$meta.id"
+    label 'process_low'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), meta:meta, publish_by_meta:['id']) }
+
+    //NB: You likely want to override this with a container containing all required
+    //dependencies for you analysis. The container at least needs to contain the
+    //yaml and rmarkdown R packages.
+    //TODO: what container to use as default image?
```

Comment:
`TODO: what container to use as default image?`

For most notebooks, users will have to override the conda/container directive anyway with their own environment that covers all the specific dependencies needed. The question for me is, if the default image should be a minimal stack that runs the test notebooks, or if we should use a fully-fletched data-science image (pandas, numpy, scipy etc.) that covers many notebooks out-of-the box.

Reply:
I decided to go for a minimal image and requested the corresponding biocontainers.